text/0061-error-format-and-definitions.md

```diff
@@ -0,0 +1,901 @@
+}
+```
+
+---
+
+## index_already_exists
+
+### Context
+
+This error happens when a user tries to create an index that already exists.
+
+### Error Definition
+
+HTTP Code: `409 Conflict`
+
+```json
+{
+    "message": "Index :uid already exists.",
+    "code": "index_already_exists",
+    "type": "invalid_request",
+    "link":"https://docs.meilisearch.com/errors#index_already_exists"
+}
+```
+
+- The `:uid` is inferred when the message is generated.
+
+---
+
+## invalid_index_uid
+
+### Context
+
+This error happens when a user tries to create an index with an invalid uid format.
+
+### Error Definition
+
+HTTP Code: `400 Bad Request`
+
+```json
+{
+     "message": ":uid is not a valid index uid. Index uid can be an integer or a string containing only alphanumeric characters, hyphens (-) and underscores (_).",
+    "code": "invalid_index_uid",
+    "type": "invalid_request",
+    "link": "https://docs.meilisearch.com/errors#invalid_index_uid"
+}
+```
+
+- The `:uid` is inferred when the message is generated.
+
+---
+
+## index_primary_key_already_exists
+
+### Context
+
+This error happens when a user try to update an index primary key while the index already has one primary key.
+
+### Error Definition
+
+HTTP Code: `400 Bad Request`
+
+```json
+{
+    "message": "Index :uid already has a primary key.",
+    "code": "index_primary_key_already_exists",
+    "type": "invalid_request",
+    "link": "https://docs.meilisearch.com/errors#index_primary_key_already_exists"
+}
+```
+
+---
+
+## primary_key_inference_failed
+
+### Context
+
+This error occurs when the engine does not find an identifier in the payload documents to define it as the primary key of the index during the inference process when no document has already been inserted.
+
+### Error Definition
+
+Async
+
+```json
+{
+    "message": "The primary key inference process failed because the engine did not find any fields containing `id` substring in their name. If your document identifier does not contain any `id` substring, you can set the primary key of the index.",
+    "code": "primary_key_inference_failed",
+    "type": "invalid_request",
+    "link": "https://docs.meilisearch.com/errors#primary_key_inference_failed"
+}
+```
+
+---
+
+## missing_document_id
+
+### Context
+
+This error occurs when the engine does not find the primary key previously defined for the index in the document payload.
+
+### Error Definition
+
+Async
+
+```json
+{
+    "message": "Document doesn't have a :primaryKey attribute: :documentRepresentation",
+    "code": "missing_document_id",
+    "type": "invalid_request",
+    "link": "https://docs.meilisearch.com/errors#missing_document_id"
+}
+```
+
+- The `:primaryKey` is inferred when the message is generated. This is the value of the primaryKey attribute of the related index object.
+- The `:documentRepresentation` is inferred when the message is generated.
+
+---
+
+## document_fields_limit_reached
```

Comment:
Why this one is not an `internal`, like `database_size_limit_reached`?

Reply:
It's on the user side because the user can cause this error by sending a document containing `65536` fields in a payload 🤓